pkg/cmd/pipeline/start.go

```diff
@@ -95,49 +117,223 @@ tkn pipeline start foo --param NAME=VALUE --resource source=scaffold-git  -s Ser
 			return NameArg(args, p)
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-
-			opt := startOptions{
-				Resources:          res,
-				Params:             params,
-				ServiceAccountName: svc,
-				ServiceAccounts:    svcs,
-				Last:               last,
+			opt.stream = &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
 			}
-			return startPipeline(cmd.OutOrStdout(), opt, p, args[0])
+			return opt.run(args[0])
 		},
 	}
 
-	c.Flags().StringSliceVarP(&res, "resource", "r", []string{}, "pass the resource name and ref")
-	c.Flags().StringSliceVarP(&params, "param", "p", []string{}, "pass the param")
-	c.Flags().StringVarP(&svc, "serviceaccount", "s", svc, "pass the serviceaccount name")
+	c.Flags().StringVarP(&opt.ServiceAccountName, "serviceaccount", "s", "", "pass the serviceaccount name")
 	flags.AddShellCompletion(c.Flags().Lookup("serviceaccount"), "__kubectl_get_serviceaccount")
-	c.Flags().StringSliceVar(&svcs, "task-serviceaccount", []string{}, "pass the service account corresponding to the task")
+	c.Flags().StringSliceVar(&opt.ServiceAccounts, "task-serviceaccount", []string{}, "pass the service account corresponding to the task")
 	flags.AddShellCompletion(c.Flags().Lookup("task-serviceaccount"), "__kubectl_get_serviceaccount")
-
-	c.Flags().BoolVarP(&last, "last", "l", false, "re-run the pipeline using last pipelinerun values")
+	c.Flags().BoolVarP(&opt.Last, "last", "l", false, "re-run the pipeline using last pipelinerun values")
 
 	_ = c.MarkZshCompPositionalArgumentCustom(1, "__tkn_get_pipeline")
+
 	return c
 }
 
-func startPipeline(out io.Writer, opt startOptions, p cli.Params, pName string) error {
+func (opt *startOptions) run(pName string) error {
+	if !opt.Last {
+		if err := opt.getInput(pName); err != nil {
+			return err
+		}
+	}
+
+	return opt.startPipeline(pName)
+}
+
+func (opt *startOptions) getInput(pname string) error {
+	cs, err := opt.cliparams.Clients()
+	if err != nil {
+		return err
+	}
+	pres, err := getPipelineResources(cs.Tekton, opt.cliparams.Namespace())
+	if err != nil {
+		fmt.Fprintf(opt.stream.Err, "Failed to list pipelineresources from %s namespace \n", opt.cliparams.Namespace())
```

Comment:
I just learned that today : https://github.com/golang/go/wiki/CodeReviewComments#error-strings 

(no capitalization on error strings)

Reply:
Addressed